solutions/usaco-573.mdx

```diff
@@ -11,6 +11,99 @@ author: Qi Wang
 
 <LanguageSection>
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int number_of_wins(vector<int> &bessie, vector<int> &elsie,
+	function<bool(int, int)> cmp)
+{
+	int n = size(bessie);
+	assert(size(bessie) == size(elsie));
+
+	sort(begin(bessie), end(bessie), cmp);
+	sort(begin(elsie), end(elsie), cmp);
+
+	int bpos = 0;  // keep track of index of card which bessie uses in current move
+	int wins = 0;
+
+	for (int i = 0; i < n; i++)
+	{
+		// Check if Bessie can win. If she can't, she'll just use her worst card.
+		if (cmp(bessie[bpos], elsie[i]))
+		{
+			wins++;
+			bpos++;
+		}
+	}
+
+	return wins;
+}
+
+int main()
+{
+	freopen("cardgame.in", "r", stdin);
+	freopen("cardgame.out", "w", stdout);
+
+	int n;
+	cin >> n;
+
+	map<int, int> elsie_has;
+
+	// First and second half are independent of each other. 
+	vector<int> bessie_first(n / 2), bessie_second(n / 2);
+	vector<int> elsie_first(n / 2), elsie_second(n / 2);
+
+	for (int i = 0; i < n; i++)
+	{
+		int x;
+		cin >> x;
+
+		elsie_has[x] = 1;
+
+		if (i < n / 2) 
+		{
+			elsie_first[i] = x;
+		}
+		else
+		{
+			elsie_second[i - n / 2] = x;
+		}
+	}
+
+	for (int i = 2 * n, fpos = 0, bpos = 0; i > 0; i--)
```

Comment:
what does fpos and bpos stand for?

Reply:
Just two pointers for first and second half..

Reply:
maybe use better naming? the names fpos and bpos seems kind of random

Reply:
i removed them now...